### PR TITLE
Add Version 3.0 migration guide.

### PR DESCRIPTION
**Pull request by andrewrmiller**

### CODE REVIEW

File: doc/migration.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# Hypersync SDK Migration Guide
```
Comment (rolandhyperproof) [inline doc/migration.md:1]: nit: filename could be referencing 2->3 migration so when we go 3->4 we have consistent naming

File: doc/migration.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    "@hyperproof/hypersync-models": "^5.0.0",
+    "@hyperproof/hypersync-sdk": "^3.0.0",
+    ...
+  }
+```
+
+### Imports
+
+As noted above and in the README, a number of classes, functions, types, interfaces and enums
+been moved to `@hyperproof/integration-sdk`. Hypersync SDK developers should take note of the
```
Comment (rolandhyperproof) [inline doc/migration.md:34]: could mention what they were moved from

File: doc/migration.md (hunk 1 above)
Comment (andrewrmiller) [reply doc/migration.md:1]: The idea is that there will only be one migration guide an that each version to version step will have a section.  This is consistent with other migration guides I have seen.

File: doc/migration.md (hunk 2 above)
Comment (andrewrmiller) [reply doc/migration.md:34]: Rethinking this.

File: doc/migration.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+| Logger                | Class |
+
+### Updated Execution Environment
+
+Hypersync apps in Hyperproof EU execute in an updated execution environment. A couple of
+changes need to be made to existing Hypersync apps to allow them to run in Hyperproof EU.
+
+The primary difference between the Hyperproof execution environment and the Hyperproof EU
+execution environment is the way that Hypersync apps are started. Hyperproof EU requires
+an `up` script in `package.json` which creates an initializes an HTTP server in the app.
```
Comment (rolandhyperproof) [inline doc/migration.md:51]: the updated execution environment is also planned in the future for the regular hyperproof.app, may be worth mentioning that as well so people can prep for it.

**Review by rolandhyperproof — DISMISSED** [review]

File: doc/migration.md (hunk 3 above)
Comment (andrewrmiller) [reply doc/migration.md:51]: Think our plan needs to become more firm before I say anything about that.

File: doc/migration.md (hunk 2 above)
Comment (andrewrmiller) [reply doc/migration.md:34]: We're going a different direction here.

File: README.md [hunk 4]
```diff
@@ -32,6 +32,8 @@ To get started with the Hypersync SDK hop on over to the [SDK documentation](doc
 
 - Various enhancements and fixes have been made to improve overall quality and reliability.
 
+For more information on migrationg from version 2.X to 3.0, please see the [migration guide](./doc/migration.md).
```
Comment (gvillenave) [inline README.md:35]: Typo
```suggestion
For more information on migrating from version 2.X to 3.0, please see the [migration guide](./doc/migration.md).
```

File: doc/migration.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    ...
+  }
+```
+
+If your code imported classes, types, interfaces, or enums from one of the
+other packages, those imports will need to be updated after making this change.
+
+### Updated Execution Environment
+
+Hypersync apps in Hyperproof EU execute in an updated execution environment. A couple of
```
Comment (gvillenave) [inline doc/migration.md:36]: Do we want to mention FR here as well, or too soon?

File: doc/migration.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+After making these changes, when you build your app it will produce `./build/start.js` that is
+referenced by the `up` command.
+
+Once you have made this change, your custom Hypersync app will function correctly in either
+the Hyperproof or Hyperproof EU environments.
+
+> **NOTE**
+>
+> If you want to deploy your custom app to Hyperproof EU, you will need [HP CLI version](./hyperproof-cli.md)
+> version 1.1.0.0 or higher. See the `--domain` option on the `hp signin` command.
```
Comment (gvillenave) [inline doc/migration.md:83]: Is that version number correct?

**Review by gvillenave — APPROVED** [review]

File: doc/migration.md (hunk 6 above)
Comment (andrewrmiller) [reply doc/migration.md:83]: Yeah, thanks Microsoft.  Major, Minor, Patch, Version.

https://learn.microsoft.com/en-us/javascript/api/sp-core-library/version?view=sp-typescript-latest

File: doc/migration.md (hunk 5 above)
Comment (andrewrmiller) [reply doc/migration.md:36]: Too soon.  I have a work item to update this with Hyperproof Gov refs once we get there.

File: README.md (hunk 4 above)
Comment (andrewrmiller) [reply README.md:35]: Thanks for catching this.

Comment (andrewrmiller) [reply README.md]: Fixed.

**Pull request merged**